dev/tasks/crossbow.py

```diff
@@ -38,6 +42,103 @@
 CWD = Path(__file__).parent.absolute()
 
 
+NEW_FEATURE = 'New Features and Improvements'
+BUGFIX = 'Bug Fixes'
+
+
+class JiraChangelog:
+
+    def __init__(self, version, username, password,
+                 server='https://issues.apache.org/jira'):
+        self.server = server
+        # clean version to the first numbers
+        self.version = '.'.join(version.split('.')[:3])
+        query = ("project=ARROW "
+                 "AND fixVersion='{0}' "
+                 "AND status = Resolved "
+                 "AND resolution in (Fixed, Done) "
+                 "ORDER BY issuetype DESC").format(self.version)
+        self.client = jira.client.JIRA({'server': server},
+                                       basic_auth=(username, password))
+        self.issues = self.client.search_issues(query, maxResults=9999)
+
+    def format_markdown(self):
+        out = StringIO()
+
+        issues_by_type = toolz.groupby(lambda i: i.fields.issuetype.name,
+                                       self.issues)
+        for typename, issues in sorted(issues_by_type.items()):
+            issues.sort(key=lambda x: x.key)
+
+            out.write('## {0}\n\n'.format(typename))
+            for issue in issues:
+                out.write('* {0} - {1}\n'.format(issue.key,
+                                                 issue.fields.summary))
```

Comment:
The summary would need escaping to avoid producing Markdown syntax (e.g. underscores & asterisks).